src/Data/Char/Unicode.purs

```diff
@@ -478,41 +487,43 @@ isSymbol c =
 
 -- | Convert a letter to the corresponding upper-case letter, if any.
 -- | Any other character is returned unchanged.
-toUpper :: Char -> Char
-toUpper = fromCharCode <<< uTowupper <<< toCharCode
+toUpper :: CodePoint -> CodePoint
+toUpper = modify uTowupper
```

Comment:
`toUpper` can't be `CodePoint -> CodePoint`, it must be `CodePoint -> String`. But maybe that's something we should address separately, since I see it's currently broken anyway.

Reply:
Okay ... Is this because a some precomposed characters don't have direct upper case equivalents and would need to be decomposed?

Reply:
Possibly, but also see https://unicode.org/faq/casemap_charprop.html#11

Reply:
Okay, cool ... I'm not familiar with the representation of the rules, so I wouldn't know how to fix that. Also, would we want to return `String` or `Array CodePoint`? (maybe the latter for the the `Internal` module?)

Reply:
`Array CodePoint` sounds good.

Reply:
currently it looks like `toUpper` does not change the eszett.

Reply:
@MonoidMusician could you also add some documentation to `toUpper` explaining why the type is `CodePoint -> Array CodePoint`?

Also, I'm wondering if it would be helpful to have a function like `unsafeToUpper :: CodePoint -> CodePoint` for people who want to easily map over a list of codepoints, without worrying about being 100% correct.  Its use should be discouraged of course.